docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,230 @@
+
+## UNC paths
+
+Universal naming convention (UNC) paths, which are used to access network resources, have the following format:
+
+- A server or host name, which is prefaced by \\\\. The server name can be a NetBIOS machine name or an IP/FQDN address (IPv4 as well as v6 are supported).
+- A share name, which is separated from the host name by \\. Together, the server and share name make up the volume.
+- A directory name. The [directory separator character](<xref:System.IO.Path.DirectorySeparatorChar>) separates subdirectories within the nested directory hierarchy.
+- An optional filename. The [directory separator character](<xref:System.IO.Path.DirectorySeparatorChar>) separates the file path and the filename.
+
+The following are some examples of UNC paths:
+
+|Path  |Description  |
+| -- | -- |
+| `\\system07\C$\` | The root directory of the C: drive on `system07`. |
+| `\\Server2\Share\Test\Foo.txt` | The Foo.txt file in the Test directory of the \\\\Server2\\Share volume.|
+
+UNC paths must always be fully qualified. They can include relative directory segments (`.` and `..`), but these must be part of a fully qualified path. You can use relative paths only by mapping a UNC path to a drive letter.
+
+## DOS device paths
+
+The Windows operating system has a unified object model that points to all resources, including files. These object paths are accessible from the console window and are exposed to the Win32 layer through a special folder of symbolic links that legacy DOS and UNC paths are mapped to. This special folder is accessed via the DOS device path syntax, which is one of:
+
+`\\.\C:\Test\Foo.txt`  
+`\\?\C:\Test\Foo.txt`
+
+> [!NOTE]
+> DOS device path syntax is supported on .NET implementations running on Windows starting with .NET Core 1.1 and .NET Framework 4.6.2.
+
+The DOS device path consists of the following components:
+
+- The device path specifier (`\\.\` or `\\?\`), which identifies the path as a DOS device path.
+
+   > [!NOTE]
+   > The `\\?\` is supported in all versions of .NET Core and in the .NET Framework starting with version 4.6.2.
+   
+- A symbolic link to the "real" device object (C: in this case).
+
+   The first segment of the DOS device path after the device path specifier identifies the volume or drive. (For example, `\\?\C:\` and `\\.\BootPartition\`.)
+
+   There is a specific link for UNCs that is called, not surprisingly, `UNC`. For example:
+
+      `\\.\UNC\Server\Share\Test\Foo.txt`
+      `\\?\UNC\Server\Share\Test\Foo.txt`
+
+    For device UNCs, the server/share portion is forms the volume. For example, in `\\?\server1\e:\utilities\\filecomparer\`, the server/share portion is server1\utilities. This is significant when calling a method such as <xref:System.IO.Path.GetFullPath(System.String,System.String)?displayProperty=nameWithType> with relative directory segments; it is never possible to navigate past the volume. 
```

Comment:
cc: @JeremyKuhne @rpetrusha @BillWagner 

What is `e:\` doing here in the middle of... `\\?\server1\e:\utilities\\filecomparer\`?  And where is the `\\?\UNC` that this example is supposed to be demonstrating? Also, is the double slash in the middle of `utilities\\filecomparer` legit? If so, what is its semantic role?

These seem like errors, which are currently active on the docs page "**[File path formats on Windows systems](https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats)**":

https://github.com/dotnet/docs/blob/151764c03161a98809d2e08b7368c9e8f0f22c5b/docs/standard/io/file-path-formats.md#L98


Perhaps it should read as follows instead:

> For device UNCs, the server/share portion forms the volume. For example, in `\\?\UNC\server1\utilities\filecomparer\`, the server/share portion is "`server1\utilities`", which corresponds to the UNC share `\\server1\utilities`.

As long as we're talking about the **DOS Device Paths** part of this page, the structure of this small section needs fixing in general. The two *bullet points* are a fine idea, but the second one wanders off. The paragraph beginning with "The first segment of the DOS device path after the device path specifier..." should be ***outdented***, because it is simply repeating the larger context by summarizing *both* bullet points (in fact, the whole section writ large). Also, the word "segment" there seems to be used without antecedent, or needs definition.

If the UNC information was supposed to be a specific variant of the "volume or drive" "segment" only (i.e., the second bullet point), it muddys the waters by then returning to full DOS Device Path examples (i.e., which include the "device path specifier").  So the whole UNC section should probably be outdented in general.

Finally, I was thinking there could be a third bullet point, corresponding to any (optional) additional "segments" that may follow after the (i.) "device path specifier" (segment?) and (ii.) its "***first*** [subsequent] segment [identifying] the volume or drive" (which could be a "server/share portion" instead). The putuative bullet (iii.) would clarify that some devices allow a third "segment" of any further path navigation into the device, but this seems to get beyond the scope. So maybe easier to just delete the word "first."